codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs

```diff
@@ -1521,26 +1521,38 @@ insertTypeNames names =
         |]
 
 -- | Get the list of a term names in the root namespace according to the name lookup index
-rootTermNames :: Transaction [NamedRef (Referent.TextReferent, Maybe NamedRef.ConstructorType)]
-rootTermNames = do
-  (fmap . fmap) unRow <$> queryListRow_ sql
+rootTermNamesByPath :: Maybe Text -> Transaction ([NamedRef (Referent.TextReferent, Maybe NamedRef.ConstructorType)], [NamedRef (Referent.TextReferent, Maybe NamedRef.ConstructorType)])
+rootTermNamesByPath mayNamespace = do
+  let (namespace, subnamespace) = case mayNamespace of
+        Nothing -> ("", "*")
+        Just namespace -> (namespace, globEscape namespace <> ".*")
```

Comment:
I wonder if we should escape `namespace` too, in case we go back to allowing things like `Int.*.doc`? In fact, I wonder if we should escape `.` characters in name segments for that reason, too.

Reply:
`namespace` isn't being passed to a `GLOB`, so there's no need to escape it, or did you mean something else?